misc/python/materialize/parallel_workload/worker.py

```diff
@@ -0,0 +1,59 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import random
+import time
+from typing import List, Tuple
+
+import pg8000
+
+from materialize.parallel_workload.action import Action
+
+
+class Worker:
+    rng: random.Random
+    actions: List[Tuple[Action, int]]
+    end_time: float
+    num_queries: int
+    autocommit: bool
+
+    def __init__(
+        self,
+        rng: random.Random,
+        actions: List[Tuple[Action, int]],
+        end_time: float,
+        autocommit: bool,
+    ):
+        self.rng = rng
+        self.actions = actions
+        self.end_time = end_time
+        self.num_queries = 0
+        self.autocommit = autocommit
+
+    def run(self, host: str, port: int, database: str) -> None:
+        self.conn = pg8000.connect(
+            host=host, port=port, user="materialize", database=database
+        )
+        # TODO: Transaction isolation level
```

Comment:
You may wish to generate a SetTransactionIsolationLevel action periodically (with very low probability) instead of selecting it just once at startup.